src/tests/test_frodokem.cpp

```diff
@@ -0,0 +1,224 @@
+
+      Test::Result run_one_test(const std::string& header, const VarMap& vars) override {
+         Test::Result result(Botan::fmt("FrodoKEM KAT {}", header));
+
+         const auto mode = get_mode(header);
+         const Botan::FrodoKEMConstants consts(mode);
+
+         // Our implementation performs three independent RNG invocations to get
+         // the seeds (s, seed_sk and z). The reference implementation assumes one
+         // concatenated RNG invocation and slices the seeds from a single buffer.
+         // We have to emulate this behaviour with the Fixed_Output_RNG below.
+         CTR_DRBG_AES256 ctr_drbg(vars.get_req_bin("seed"));
+         Fixed_Output_RNG fixed_generation_rng(ctr_drbg.random_vec<std::vector<uint8_t>>(
+            consts.len_sec_bytes() + consts.len_se_bytes() + consts.len_a_bytes()));
+
+         // Key generation
+         Botan::FrodoKEM_PrivateKey sk(fixed_generation_rng, mode);
+         result.test_is_eq("Generated private key", shake256_16(sk.raw_private_key_bits()), vars.get_req_bin("sk"));
+
+         auto pk = sk.public_key();
+         result.test_is_eq("Generated public key", shake256_16(pk->public_key_bits()), vars.get_req_bin("pk"));
+
+         // Encapsulation
+         Botan::FrodoKEM_PublicKey pk2(pk->public_key_bits(), mode);
+         Fixed_Output_RNG fixed_encapsulation_rng(
+            ctr_drbg.random_vec<std::vector<uint8_t>>(consts.len_sec_bytes() + consts.len_salt_bytes()));
+         auto enc = Botan::PK_KEM_Encryptor(pk2, "Raw");
+         const auto encaped = enc.encrypt(fixed_encapsulation_rng, 0 /* no KDF */);
+         result.test_is_eq("Shared Secret", encaped.shared_key(), Botan::lock(vars.get_req_bin("ss")));
+         result.test_is_eq("Ciphertext", shake256_16(encaped.encapsulated_shared_key()), vars.get_req_bin("ct"));
+
+         // Decapsulation
+         Botan::FrodoKEM_PrivateKey sk2(sk.private_key_bits(), mode);
+         Botan::Null_RNG null_rng;
+         auto dec = Botan::PK_KEM_Decryptor(sk2, null_rng, "Raw");
+         const auto shared_key = dec.decrypt(encaped.encapsulated_shared_key(), 0 /* no KDF */);
+         result.test_is_eq("Decaps. Shared Secret", shared_key, Botan::lock(vars.get_req_bin("ss")));
+
+         return result;
+      }
+};
+   #endif
+
+std::vector<Test::Result> test_frodo_roundtrips() {
+   auto& rng = Test::rng();
+
+   auto modes = std::vector {
+      Botan::FrodoKEMMode::eFrodoKEM1344_SHAKE, Botan::FrodoKEMMode::eFrodoKEM976_SHAKE,
+         Botan::FrodoKEMMode::eFrodoKEM640_SHAKE, Botan::FrodoKEMMode::FrodoKEM1344_SHAKE,
+         Botan::FrodoKEMMode::FrodoKEM976_SHAKE, Botan::FrodoKEMMode::FrodoKEM640_SHAKE,
+   #if defined(BOTAN_HAS_AES)
+         Botan::FrodoKEMMode::eFrodoKEM1344_AES, Botan::FrodoKEMMode::eFrodoKEM976_AES,
+         Botan::FrodoKEMMode::eFrodoKEM640_AES, Botan::FrodoKEMMode::FrodoKEM1344_AES,
+         Botan::FrodoKEMMode::FrodoKEM976_AES, Botan::FrodoKEMMode::FrodoKEM640_AES
+   #endif
```

Comment:
Instead of excluding the modes at compile time in this way lets just `continue` if `!mode.is_supported()`